### PR TITLE
feat(reports): add reporters and basic first reports

### PR DESCRIPTION
**Pull request by TheMaskedTurtle**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**
Add reporters to the OpenReac API and first basic reports on the number of elements in the parameters.

**What is the current behavior?**
No reporters at all.

**What is the new behavior (if this is a feature change)?**
We add the possibility to use reporters to the API without breaking it and produce the first reports.

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [x] No

### CODE REVIEW

File: open-reac/src/main/java/com/powsybl/openreac/Reports.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+/**
+ * @author Joris Mancini <joris.mancini_externe at rte-france.com>
+ */
+public final class Reports {
+
+    private Reports() {
+        // Should not be instantiated
+    }
+
+    public static Reporter createVoltageInitReporter(Reporter reporter, String networkId) {
```
Comment (annetill) [inline open-reac/src/main/java/com/powsybl/openreac/Reports.java:22]: Please remove any reference to voltage init or voltage initialisation : this is linked to our legacy. Here it is called OpenReac. Maybe adding a reference to the objective function is needed to fit totally with our legacy.

File: open-reac/src/main/java/com/powsybl/openreac/Reports.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        return reporter.createSubReporter("voltageInit", "Voltage init on network '${networkId}'", "networkId", networkId);
+    }
+
+    public static void reportConstantQGeneratorsSize(Reporter reporter, int constantQGeneratorsSize) {
+        reporter.report(Report.builder()
+            .withKey("constantQGeneratorsSize")
+            .withDefaultMessage("Reactive power set point is considered fixed for ${size} generators")
```
Comment (annetill) [inline open-reac/src/main/java/com/powsybl/openreac/Reports.java:29]: Reactive Target is better to fit with Powsybl wording.

File: open-reac/src/main/java/com/powsybl/openreac/Reports.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            .withSeverity(TypedValue.INFO_SEVERITY)
+            .withValue("size", constantQGeneratorsSize)
+            .build());
+    }
+
+    public static void reportVariableTwoWindingsTransformersSize(Reporter reporter, int variableTwoWindingsTransformersSize) {
+        reporter.report(Report.builder()
+            .withKey("variableTwoWindingsTransformersSize")
+            .withDefaultMessage("${size} two-windings transformers are considered as variable")
```
Comment (annetill) [inline open-reac/src/main/java/com/powsybl/openreac/Reports.java:38]: variable tap positions? Same for shunt variable shunt sections?

Comment (TheMaskedTurtle) [reply open-reac/src/main/java/com/powsybl/openreac/Reports.java:38]: @annetill like "There are ${size} tap positions considered as variable on two-winding transformers" and "There are ${size} shunt sections considered as variable on shunt compensators" or just "There are ${size} tap positions considered as variable" and "There are ${size} shunt sections considered as variable" ?

File: open-reac/src/main/java/com/powsybl/openreac/Reports.java (hunk 1 above)
Comment (TheMaskedTurtle) [reply open-reac/src/main/java/com/powsybl/openreac/Reports.java:22]: Sure, will do !

File: open-reac/src/main/java/com/powsybl/openreac/Reports.java (hunk 2 above)
Comment (TheMaskedTurtle) [reply open-reac/src/main/java/com/powsybl/openreac/Reports.java:29]: Reactive power target or just reactive target ?

Comment (annetill) [reply open-reac/src/main/java/com/powsybl/openreac/Reports.java:29]: Reactive power target

File: open-reac/src/main/java/com/powsybl/openreac/Reports.java (hunk 3 above)
Comment (annetill) [reply open-reac/src/main/java/com/powsybl/openreac/Reports.java:38]: First proposals.

File: open-reac/src/main/java/com/powsybl/openreac/Reports.java (hunk 2 above)
Comment (TheMaskedTurtle) [reply open-reac/src/main/java/com/powsybl/openreac/Reports.java]: Ok, fixed.

**Review by annetill — APPROVED** [review]

**Pull request merged**